CDEFramework/Libraries/LegacyDictionarySupport/HttpHandlers/GlossaryLinkHrefHandler.cs

```diff
@@ -0,0 +1,208 @@
+﻿using System;
+using System.Web;
+using System.Net.Http;
+using System.Collections.Generic;
+using System.Configuration;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+
+using LegacyDictionarySupport.Configuration;
+
+namespace LegacyDictionarySupport.HttpHandlers
+{
+
+    public class GlossaryLinkHrefHandler : IHttpHandler
+    {
+        // implement IsReusable method
+        public bool IsReusable
+        {
+            get { return false; }
+        }
+
+        public void ProcessRequest(HttpContext context)
+        {
+            HttpRequest request = context.Request;
+
+            // Format ID parameter for API call
+            string id = request.QueryString["id"];
+            id = Regex.Replace(id, "^CDR0+", "", RegexOptions.Compiled);
+
+            // Format dictionary and audience parameters for API call
+            string dictionary = request.QueryString["dictionary"];
+            if (!string.IsNullOrWhiteSpace(dictionary) && dictionary.ToLower() == "genetic")
+            {
+                dictionary = "Genetics";
+            }
+
+            string audience = request.QueryString["version"];
+            if (!string.IsNullOrWhiteSpace(audience) && audience.ToLower() == "patient")
+            {
+                audience = "Patient";
+            }
+            if (!string.IsNullOrWhiteSpace(audience) && audience.ToLower() == "healthprofessional")
+            {
+                audience = "HealthProfessional";
+            }
+
+            if (string.IsNullOrWhiteSpace(dictionary) && string.IsNullOrWhiteSpace(audience))
+            {
+                dictionary = "Cancer.gov";
+                audience = "Patient";
+            }
+
+            if (!string.IsNullOrWhiteSpace(audience) && string.IsNullOrWhiteSpace(dictionary))
+            {
+                switch (audience.ToLower())
+                {
+                    case "patient":
+                        dictionary = "Cancer.gov";
+                        audience = "Patient";
+                        break;
+
+                    case "healthprofessional":
+                        dictionary = "NotSet";
+                        audience = "HealthProfessional";
+                        break;
+
+                    default:
+                        dictionary = "NotSet";
+                        audience = "Patient";
+                        break;
+                }
+            }
+
+            if (!string.IsNullOrWhiteSpace(dictionary) && string.IsNullOrWhiteSpace(audience))
+            {
+                audience = dictionary.ToLower() == "cancer.gov" ? "Patient" : "HealthProfessional";
+            }
```

Comment:
This block is confusing.  At a minimum, please add a comment block explaining the detailed _intent_, because trying to read through it, it feels as though you're potentially setting `dictionary` and `audience` multiple times, potentially to contradictory values.

Potentially, there should be separate functions for getting the `dictionary` and `audience` values.

Reply:
I see you've made some changes, but what's still missing is a detailed comment to _explain the business rule you're implementing_.  That's what I meant about the intent.